src/HypercertExchangeClient.ts

```diff
@@ -435,18 +453,61 @@ export class HypercertExchangeClient {
     return approve(signer, tokenAddress, spenderAddress, amount, overrides);
   }
 
+  /**
+   * Approve an ERC20 to be used as a currency on the Hypercert Exchange using Safe
+   * @param tokenAddress Address of the ERC20 to approve
+   * @param amount Amount to be approved (default to MaxUint256)
+   * @param safeAddress Address of the Safe to use
+   * @returns Safe transaction hash
+   */
+  public approveErc20Safe(
+    safeAddress: string,
+    tokenAddress: string,
+    amount: bigint = MaxUint256,
+    overrides?: Overrides
+  ): Promise<string> {
+    if (!this.walletClient) {
+      throw new Error("No wallet client");
+    }
+
+    const safeTransactionBuilder = new SafeTransactionBuilder(this.walletClient, this.chainId, this.addresses);
+    return safeTransactionBuilder.approveErc20(safeAddress, tokenAddress, amount);
+  }
+
   /**
    * Check whether or not an operator has been approved by the user
    * @param operator Operator address (default to the exchange address)
    * @returns true if the operator is approved, false otherwise
    */
   public async isTransferManagerApproved(
     operator: string = this.addresses.EXCHANGE_V2,
+    signerAddress?: string,
     overrides?: Overrides
   ): Promise<boolean> {
     const signer = this.getSigner();
-    const signerAddress = await signer.getAddress();
-    return hasUserApprovedOperator(signer, this.addresses.TRANSFER_MANAGER_V2, signerAddress, operator, overrides);
+    const signerAddressFallback = await signer.getAddress();
```

Comment:
I think this update should make the field mandatory. It'd be a headache if you get an "is approved" response but your expected tx fails. My proposal is to make signerAddress mandatory and add and `isAddress` validation. Throw when the address is invalid

Reply:
This was actually an attempt to make the method reuseable, so you can check either for the address for the current signer, or a provided address. I ended up implementing a separate method for the safe.

I'd argue for rolling back this change, as it keeps the safe integration feature backwards compatible.

Reply:
Reverted